### DISCUSSION TITLE
change data in dynamic route

### DISCUSSION THREAD

**Discussion by shahabbasian** [nuxt/nuxt#17993] [Questions] [answered]
I have a component that receives and displays data through Props

```
<template>
  <div>
    <div v-for="item in data" :key="item.id">
      <NuxtLink :to="'/' + ${item.id}">{{ item.id }}</NuxtLink>
    </div>
  </div>
</template>

<script setup>
defineProps(['data']);
</script>
```

And I have a dynamic page:
`[id].vue`

```
<template>
  <div>
    <CategoryAll :data="data" />
  </div>
</template>

<script setup>
const route = useRoute();
const { data } = await apiGetChildCategories(route.params.id);
</script>
```

I want to see the subcategories of the category by clicking on the link, and again by clicking on the next subcategories ...

But this happens only for the first time and in the next clicks only the link changes but the new data is not fetch

**Comment by danielroe** [comment, marked as answer]
You can follow this issue as well: https://github.com/nuxt/framework/issues/2903
